src/inmanta_module_factory/inmanta/attribute.py

```diff
@@ -67,6 +72,12 @@ def is_list(self) -> bool:
 
     @property
     def inmanta_type(self) -> str:
+        if isinstance(self._inmanta_type, typedef.TypeDef):
```

Comment:
Given my ingrained reluctance about isinstance, I would add a `get_name_in(module: Module)`  method to the `InmantaBaseType` that return the correct name, within a module